superset/db_engine_specs/presto.py

```diff
@@ -944,30 +955,31 @@ def where_latest_partition(cls, table_name, schema, database, qry, columns=None)
         column_names = {column.get("name") for column in columns or []}
         for col_name, value in zip(col_names, values):
             if col_name in column_names:
-                qry = qry.where(Column(col_name) == value)
-        return qry
+                query = query.where(Column(col_name) == value)
+        return query
 
     @classmethod
-    def _latest_partition_from_df(cls, df):
+    def _latest_partition_from_df(cls, df) -> Optional[Tuple[str, ...]]:
         if not df.empty:
             return df.to_records(index=False)[0].item()
+        return None
 
     @classmethod
-    def latest_partition(cls, table_name, schema, database, show_first=False):
+    def latest_partition(
+        cls, table_name: str, schema: str, database, show_first: bool = False
+    ) -> Tuple[List[str], Optional[Tuple[str, ...]]]:
         """Returns col name and the latest (max) partition value for a table
 
         :param table_name: the name of the table
-        :type table_name: str
         :param schema: schema / database / namespace
-        :type schema: str
         :param database: database query will be run against
         :type database: models.Database
         :param show_first: displays the value for the first partitioning key
           if there are many partitioning keys
         :type show_first: bool
 
         >>> latest_partition('foo_table')
-        ('ds', '2018-01-01')
+        (['ds'], ('2018-01-01',))
```

Comment:
@etr2460 Can you check that I got this right? Based on my googling/testing `df.to_records(index=False)[0].item()` in `_latest_partition_from_df` returns a tuple with whatever that row contains (can strictly speaking be `Any`, but in this case I'm assuming they are always `str`)

Reply:
..cont'd. Adding typing to those methods opened up a massive can of worms that I didn't want to wrestle right now. So removed some typing from the Presto spec. Can get back to those later, but would like to prioritize sorting out the circular imports for now.

Reply:
Yeah, i'm not certain if this is right either... There's been quite a few changes to the presto db engine spec lately which have introduced bugs. Once the circular imports are sorted out it'd probably be worthwhile to come back to this; I'd be happy to pair code/work with you on this